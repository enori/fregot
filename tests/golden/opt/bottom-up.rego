package bottom_up

primes = [2, 3, 5, 7, 11, 13, 17, 19, 23, 29, 31, 37, 41, 43, 47, 53, 59, 61, 67, 71, 73, 79, 83, 89, 97]

double_primes[ret] {
  # Should be bottom up, `ret` only assigned.
  primes[p]
  ret := p * 2
}

small_primes[ret] {
  # Should not be bottom up because of the restriction on `ret`.
  ret := primes[_]
  ret < 10
}
