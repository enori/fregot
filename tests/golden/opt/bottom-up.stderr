primes = [
  2,
  3,
  5,
  7,
  11,
  13,
  17,
  19,
  23,
  29,
  31,
  37,
  41,
  43,
  47,
  53,
  59,
  61,
  67,
  71,
  73,
  79,
  83,
  89,
  97
]
[bottom-up double_primes]
double_primes [ret] {
  bottom_up.primes.p
  ret := *(p, 2)
}
small_primes [ret] {
  ret := bottom_up.primes._
  <(ret, 10)
}
